manual/src/cmds/tsan.etex

```diff
@@ -0,0 +1,227 @@
+    #2 caml_start_program <null> (simple_race.exe+0x47cf37)
+    #3 caml_callback_exn runtime/callback.c:197 (simple_race.exe+0x445f7b)
+    #4 domain_thread_func runtime/domain.c:1167 (simple_race.exe+0x44a113)
+
+  Previous read of size 8 at 0x8febe0 by main thread (mutexes: write M86):
+    #0 camlSimple_race.d1_271 simple_race.ml:5 (simple_race.exe+0x420a22)
+    #1 camlSimple_race.entry simple_race.ml:13 (simple_race.exe+0x420d16)
+    #2 caml_program <null> (simple_race.exe+0x41ffb9)
+    #3 caml_start_program <null> (simple_race.exe+0x47cf37)
+[...]
+
+WARNING: ThreadSanitizer: data race (pid=3808831)
+  Read of size 8 at 0x8febf0 by thread T1 (mutexes: write M90):
+    #0 camlSimple_race.d2_274 simple_race.ml:9 (simple_race.exe+0x420a92)
+    #1 camlDomain.body_706 stdlib/domain.ml:211 (simple_race.exe+0x440f2f)
+    #2 caml_start_program <null> (simple_race.exe+0x47cf37)
+    #3 caml_callback_exn runtime/callback.c:197 (simple_race.exe+0x445f7b)
+    #4 domain_thread_func runtime/domain.c:1167 (simple_race.exe+0x44a113)
+
+  Previous write of size 8 at 0x8febf0 by main thread (mutexes: write M86):
+    #0 camlSimple_race.d1_271 simple_race.ml:4 (simple_race.exe+0x420a01)
+    #1 camlSimple_race.entry simple_race.ml:13 (simple_race.exe+0x420d16)
+    #2 caml_program <null> (simple_race.exe+0x41ffb9)
+    #3 caml_start_program <null> (simple_race.exe+0x47cf37)
+[...]
+
+==================
+ThreadSanitizer: reported 2 warnings
+\end{verbatim}
+
+For each detected data race, TSan reports the location of the conflicting
+accesses, their nature (read, write, atomic read, etc.), and the associated
+stack trace.
+
+If you run the above program several times, the output may vary: sometimes TSan
+will report two data races, sometimes one, and sometimes none. This is due to
+the combination of two factors:
+
+\begin{itemize}
+  \item First, TSan reports only the data races encountered during execution,
+    i.e., conflicting, unordered memory accesses that are effectively performed.
+  \item In addition, in this program, depending on executions, there may be no
+    such memory accesses: if \texttt{d1} returns before \texttt{d2} has
+    finished spawning, then all memory accesses originating from \texttt{d1}
+    may happen-before the ones originating from \texttt{d2}, since spawning a
+    domain involves some inter-thread synchronization. In that case, the
+    accesses are considered to be ordered and no data race is reported.
+\end{itemize}
+
+This example illustrates the fact that data races can sometimes be hidden by
+unrelated synchronizing operations.
+
+\section{s:tsan-performance}{Performance implications}
+
+TSan instrumentation imposes a non-negligible cost at runtime. Empirically,
+this cost has been observed to cause a slowdown which can range from 2x to 7x.
+One of the main factors of high slowdowns is frequent of access to mutable
+data (in contrast, reads from immutable data are not instrumented). The memory
```

Comment:
Avoid the brackets and add a separate sentence. 

"In contrast, the initialising writes to and reads from immutable memory locations are not instrumented".